app/controllers/amends/CheckYourAnswersController.scala

```diff
@@ -122,11 +129,28 @@ class CheckYourAnswersController @Inject() (
 
         returnsConnector.amend(pptId, submissionId).flatMap {
           case Right(optChargeRef) =>
-            sessionRepository.set(Entry(request.cacheKey, optChargeRef)).map{
+            sessionRepository.set(Entry(request.cacheKey, optChargeRef)).map {
               _ => Redirect(routes.AmendConfirmationController.onPageLoad())
             }
           case Left(error) =>
             throw error
         }
     }
+
+  def cancel(): Action[AnyContent] =
+    (identify andThen getData andThen requireData).async {
+      implicit request =>
+        val pptReference: String = request.request.pptReference
+        val maybePeriodKey = request.userAnswers.get(AmendSelectedPeriodKey)
+        val futureNextPage = request.userAnswers
+          .reset
+          .save(cacheConnector.saveUserAnswerFunc(pptReference))
+          .map { _ =>
+            maybePeriodKey match {
+              case Some(periodKey) => routes.ViewReturnSummaryController.onPageLoad(periodKey)
```

Comment:
Should the Cancel redirect to a "are you sure you want to cancel the amend?" page or something similar? According to Adam and Rebecca this should be the case. But please ignore this message if you have plan to do this as part of a new task.